zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/schema/SchemaManager.java

```diff
@@ -73,6 +75,40 @@ public void startup() {
     LOG.info("Schema management completed.");
   }
 
+  private void updateSchemaSettings() {
+    final var existingTemplateNames =
+        searchEngineClient
+            .getMappings(config.getIndex().getPrefix() + "*", MappingSource.INDEX_TEMPLATE)
+            .keySet();
+
+    final var existingIndexNames =
+        searchEngineClient.getMappings(allIndexNames(), MappingSource.INDEX).keySet();
+
+    indexTemplateDescriptors.stream()
+        .filter(desc -> existingTemplateNames.contains(desc.getTemplateName()))
+        .forEach(
+            desc -> {
+              searchEngineClient.updateIndexTemplateSettings(
+                  desc, getIndexSettings(desc.getIndexName()));
+            });
+
+    // update matching index for the index template descriptor
+    indexTemplateDescriptors.stream()
+        .filter(desc -> existingIndexNames.contains(desc.getFullQualifiedName()))
+        .forEach(this::updateIndexReplicaCount);
+
+    indexDescriptors.stream()
+        .filter(desc -> existingIndexNames.contains(desc.getFullQualifiedName()))
+        .forEach(this::updateIndexReplicaCount);
+  }
+
+  private void updateIndexReplicaCount(final IndexDescriptor indexDescriptor) {
+    final var indexSettings = getIndexSettings(indexDescriptor.getIndexName());
```

Comment:
🔧 I feel we could also simply do:

```
        config
            .getIndex()
            .getReplicasByIndexName()
            .getOrDefault(indexName, config.getIndex().getNumberOfReplicas());
```

Instead of constructing the settings object with unnecessary values, etc. You can also put this into an own method to get the replicaSetting from config, and use it in both places.